ui/src/components/Canvas.tsx

```diff
@@ -520,6 +523,19 @@ const CodeNode = memo<Props>(function ({
     }
   }, [xPos, yPos, setPodPosition, id]);
 
+  useEffect(() => {
+    // get relative position
+    const node = nodesMap.get(id);
+    const parentNode = nodesMap.get(node?.parentNode??'');
+    // console.log(node, parentNode)
+    if(node?.parentNode && !node.position.x && dragging){
+      node.parentNode = undefined
+      node.data!.parent = 'ROOT'
+      setPodParent({ id, parent: 'ROOT', dirty: false });
+    }
```

Comment:
Two issues:
1. this allows only dragging a pod out of a scope from **left** side, but not other 3 sides. This behavior sounds inconsistent.
2. we need to update the position (which is relative to the parent scope), otherwise, the pod will "jump" upon moving out.

More broadly speaking, we probably want another way to dissociate the pod from the scope, other than simply dragging, because the current pod-cannot-be-moved-out behavior is useful. We could probably add a "release button" which, once clicked, temporarily remove `extent: 'parent'` field to allow it to be moved out.